All_home_works/hw3/task2.py

```diff
@@ -0,0 +1,36 @@
+"""
+Here's a not very efficient calculation function that calculates something important:
+
+import time
+import struct
+import random
+import hashlib
+
+def slow_calculate(value):
+
+    time.sleep(random.randint(1,3))
+    data = hashlib.md5(str(value).encode()).digest()
+    return sum(struct.unpack('<' + 'B' * len(data), data))
+
+Calculate total sum of slow_calculate() of all numbers starting from 0 to 500.
```

Comment:
So, I have 4 core. That's why time at first not increased. As all cores worked simultaneously. Then time began to slow down with increasing processes because the core always had to do many switches from one activity to another.
![image](https://user-images.githubusercontent.com/70700249/116001104-9465b580-a5fb-11eb-9c58-f7ffa2ef306f.png)

Reply:
How can we set a proper number of cores for the execution on different machines?

Reply:
You have answered another comment.  This makes our conversation a bit messy.  Don't forget to answer to initial comment too.

Reply:
How many numbers do you need to calculate? 
From 0 to 500 -> 501 number or what? Sorry, I do not understand you.

How can we set a proper number of cores for the execution on different machines?
os.cpu_count() return amount of cores in our system

Reply:
>  From 0 to 500 -> 501 number or what? Sorry, I do not understand you.

This is a leading question. In that case, I want to bring your attention to the usage of that number. 

> os.cpu_count() return amount of cores in our system

Is it possible to use it?  (This is another leading questions, in this time it is "call to action")

Reply:
Totally did not understand you. Not needed to use 500 or 501 calculations. Needed to use as many cores as cores in the system and to checking cores by os.cpu_count(). Am I right?

Reply:
> Not needed to use 500 or 501 calculations.

What is written in the requirements about that?

> Needed to use as many cores as cores in the system and to checking cores by os.cpu_count(). Am I right?

Yes.

Reply:
But the task is: "Calculate the total sum of slow_calculate() of all numbers starting from 0 to 500."

And one more line: "Calculation time should not take more than a minute."

Even without multiprocessing, I will get a result from 0 to os.cpu_count() for this time...

Reply:
As always, I do not understand. That you speaking about Pool. My system works fine with 500 pools, but I will make as you said.

Reply:
> Even without multiprocessing, I will get a result from 0 to os.cpu_count() for this time...

The number of calculations and the number of cores are not related.  They are different arguments and changed for different reasons.  In some cases it might look like they are related.